### PR TITLE
new post: waku product update

### PR DESCRIPTION
**Pull request by D4nte**
Aim of the post:
- deprecate Waku Connect
- introduce Waku Product: why & what
- highlight Waku is multiplatform
- introduce operator outreach
- readers can expect more documentation, consolidated effort

TODO:
- [x] get some inputs (will comment)
- [x] fix the links
- [x] might re-write conclusion, not sure how happy I am with it
- [ ] Create and add forum thread

### CODE REVIEW

File: research/2022-08-01-introducing-waku-product.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+it is now easier for those interested in supporting and using the Waku network to run their own nodes.
+
+We are also [dogfooding Waku RLN](todo), a novel economic spam protection protocol
+and looking at [incentivizating the Waku Store protocol](todo),
+these will be added reasons as for why one might want to participate in the Waku network.
+
+# In Conclusion
+
+We are committed to make Waku, the communication layer of Web3, a success.
+For this endeavour, the Vac program is focusing on inventing and improving the Waku protocols (among other [topics](todo)).
```
Comment (D4nte) [inline research/2022-08-01-introducing-waku-product.md:83]: @oskarth Are you happy with this phrasing? I am not really, but running out of idea juice.

File: research/2022-08-01-introducing-waku-product.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Going mobile? Use Waku React Native.
+C++ Desktop Game? Use Waku C-Bindings.
+And of course, web app? Use js-waku.
+
+We are also consolidating the documentation for all implementations on a single website (work in progress)
+so it is easier for a developer to find the right information for their needs.
+
+We call this effort to push Waku to be used by other projects _platform outreach_.
+
+This year, we will also start the _operator outreach_ effort to push for operators to run Waku nodes.
```
Comment (D4nte) [inline research/2022-08-01-introducing-waku-product.md:72]: @jm-clius Are you happy with how I introduce the operator outreach effort?

File: research/2022-08-01-introducing-waku-product.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+As nwaku's [documentation](todo), [stability](todo) and [performance](todo) has improved,
+it is now easier for those interested in supporting and using the Waku network to run their own nodes.
+
+We are also [dogfooding Waku RLN](todo), a novel economic spam protection protocol
```
Comment (D4nte) [inline research/2022-08-01-introducing-waku-product.md:76]: @staheri14 what would be the best link if someone is keen to test RLN in the current state?

**Comment by D4nte** [comment]
Sorry for false pinging, I added reviewers but then decided against it as it is still draft.

File: research/2022-08-01-introducing-waku-product.md (hunk 2 above)
Comment (jm-clius) [reply research/2022-08-01-introducing-waku-product.md:83]: Perhaps sticking with "operator" throughout the sentence may sound like some technical specialisation is required, whereas theoretically any users can run their own nodes. How about "the _operator outreach_ effort to push for users to run their own Waku nodes."?

File: research/2022-08-01-introducing-waku-product.md (hunk 3 above)
Comment (fryorcraken) [reply research/2022-08-01-introducing-waku-product.md]: @staheri14 I included a link to a RLN github issue, let me know if there is a better entrypoint.

**Comment by fryorcraken** [comment]
@richard-ramos @jm-clius Let me know if there are note worthy ongoing/planned efforts for nwaku and go-waku that I missed.

**Comment by fryorcraken** [comment]
Note: cannot be merged until I add the forum todo link, so just comment/approve and I will then update and merge.

File: research/2022-08-20-introducing-waku-product.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+image: /img/black-waku-logo-with-name.png
+discuss: TODO
+---
+
+# Waku for Web DApps
+
+In 2021, we started to push for the adoption of Waku by projects other than the Status app.
+Waku is the communication component of the Web3 trifecta,
+which originally was Ethereum (contracts), Swarm (storage) and Whisper (communication).
+Hence, it made sense to first target dApps which already uses one of the Web3 pillar: Ethereum.
```
Comment (jm-clius) [inline research/2022-08-20-introducing-waku-product.md:20]:
```suggestion
Hence, it made sense to first target dApps which already uses one of the Web3 pillars: Ethereum. 
```

File: research/2022-08-20-introducing-waku-product.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+Finally, (3) was the most important lesson.
+We learned that complex platforms which include mobile, web, backend and desktop software
+need Waku for decentralized and censorship-resistant communications.
+For these projects, js-waku is simply not enough.
+Said projects need to integrate Waku in their Golang or NodeJS backend, in their C# desktop application and React Native mobile app.
+
+As we tackled the effort to make Waku [multi](https://github.com/status-im/go-waku/tree/master/examples)-[platform](https://github.com/status-im/waku-react-native),
+we realized we had to pivot the outreach effort beyond web dApps.
+
+We understood that we should see the whole Waku software auite (js-waku, nwaku, go-waku) as an asset for the success of Waku.
```
Comment (jm-clius) [inline research/2022-08-20-introducing-waku-product.md:60]:
```suggestion
We understood that we should see the whole Waku software suite (js-waku, nwaku, go-waku) as an asset for the success of Waku.
```

File: research/2022-08-20-introducing-waku-product.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+We are also consolidating the documentation for all implementations on a single website (work in progress)
+so it is easier for a developer to find the right information.
+
+We call the task of pushing for Waku to be used by other projects _platform outreach_.
+
+This year, we will also start the _operator outreach_ effort to push for users to run their own Waku nodes.
+Nwaku's documentation, stability and performance has improved,
+it is now easier to run your [own Waku node](https://github.com/status-im/nwaku/tree/master/docs/operators).
+
+Today, operators wannabes are most likely running their own nodes to support or use the Waku network.
```
Comment (jm-clius) [inline research/2022-08-20-introducing-waku-product.md:85]:
```suggestion
Today, operator wannabes are most likely running their own nodes to support or use the Waku network.
```

File: research/2022-08-20-introducing-waku-product.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+As we are [dogfooding Waku RLN](https://github.com/status-im/nwaku/issues/827), our novel economic spam protection protocol
```
Comment (jm-clius) [inline research/2022-08-20-introducing-waku-product.md:87]:
```suggestion
As we are [dogfooding Waku RLN](https://github.com/status-im/nwaku/issues/827), our novel economic spam protection protocol,
```

File: research/2022-08-20-introducing-waku-product.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+and looking at [incentivizating the Waku Store protocol](https://github.com/vacp2p/research/issues/99),
```
Comment (jm-clius) [inline research/2022-08-20-introducing-waku-product.md:88]:
```suggestion
and looking at [incentivizing the Waku Store protocol](https://github.com/vacp2p/research/issues/99),
```

File: research/2022-08-20-introducing-waku-product.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+What you can expect next:
+
+- [Scalability and performance studies](https://forum.vac.dev/t/waku-v2-scalability-studies/142/9) and improvement across Waku software,
+- [New websites](https://github.com/waku-org/waku.org/issues/15) to easily find documentation about Waku and its implementations,
+- Delivery of the latest Waku protocols in all code bases and cross client PoCs
+  ([noise](https://rfc.vac.dev/spec/35/), [noise-sessions](https://rfc.vac.dev/spec/37/),
+  [waku-rln-relay](https://rfc.vac.dev/spec/17/), etc),
+- Easier to [run your own waku node](https://github.com/status-im/nwaku/issues/828),
+- Improvement of existing protocols based on field-testing feedback and functional requirements
```
Comment (jm-clius) [inline research/2022-08-20-introducing-waku-product.md:106]: Perhaps useful to make it explicit here that a large focus here is on restricted devices, e.g. browsers or limited-resource devices? Many people following Waku knows that we network capacity challenges (manifesting as unreliable message delivery) and this is a major part of the work to improve Waku capacity. I'd like the community to see visible progress, so they stay on board.

**Review by jm-clius — COMMENTED** [review]
Within the scope of this post I think the important focus areas are covered. You could link to https://vac.dev/introducing-nwaku somewhere, where the `nwaku` client was introduced?

File: research/2022-08-20-introducing-waku-product.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
+---
+layout: post
+name:  "Waku Product"
```
Comment (oskarth) [inline research/2022-08-20-introducing-waku-product.md:3]: Do we want to _publicly_ call it Waku Product? As opposed to this being an internal moniker we use to make it clear what we are talking about.

From external people POV, isn't this just "Waku" (as a product?)

**Comment by oskarth** [comment]
Maybe it is just me, but I find sembr for normal text hard to read/review. Feels different from a spec where every statement has to be very precise

File: research/2022-08-20-introducing-waku-product.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+and gave [talks](https://docs.wakuconnect.dev/docs/presentations/).
+
+We also assumed that we would see patterns in the usage of Waku,
+that we would facilitate with the help of [SDKs](https://github.com/status-im/wakuconnect-vote-poll-sdk).
+
+Finally, we created a number of web apps:
+[examples](https://docs.wakuconnect.dev/docs/examples/)
+and [PoCs](https://github.com/status-iM/gnosis-safe-waku).
+
+We branded this overall web app outreach effort as Waku Connect:
```
Comment (oskarth) [inline research/2022-08-20-introducing-waku-product.md:37]: Who is the audience for this post? Do people care that we previously branded it as this?

**Comment by oskarth** [comment]
Two questions:

1) Who is the audience for this post?
2) What is the main takeaway we want people to have after reading this? Any particular action they should take?

**Review by oskarth — COMMENTED** [review]
I'm struggling a bit with he positioning of this post. It feels like a lot of internal implementation details that a lot of people don't necessarily care about.

How about reframing the post to be about:

- Waku can be used today, here's how
- We've done a lot of work to make it user-friendly for developers
-- evidence: tutorials, implementations, diff from 1y ago
- Here's what you can do now
-- target audience developers/curious observer

Less focus on "Waku Product" and "Waku Connect history branding"

- Lessons learned from not just focusing on web dapps is interesting and can be part of section like "~things we learned try to get Waku adopted" - this can be a nice aha moment for people

**Comment by oskarth** [comment]
To me:

- Target audience is developers and curious observer
- We want them to go "wow, I can use this and look at all the cool things they are doing"
- Here's what the focus is now
- Here's how I can use it and how I can help

Can also lead by saying something that Waku is trying to get main stream adoption of decentralized web, and as part of this we started a deliberate effort to make it useful and usable, and not just as a research effort.

**Comment by oskarth** [comment]
What's the state of this post @fryorcraken?

**Comment by fryorcraken** [comment]
@jm-clius @oskarth partial rewrite of the post to make it more focus on specific outcomes. Thank you for the feedback so far, it was very useful.

File: research/2022-08-20-introducing-waku-product.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+1. The potential use cases for Waku are varied and numerous,
+2. Many projects are interested in having an embedded decentralized chat feature in their dApp,
+3. There are a number of complex applications, which are not a simple web app, that need Waku as a solution.
+
+(1) means that it is not that easy to create SDKs that facilitate using js-waku's usage.
+
+(2) was a clear candidate for an SDK.
+However, building a chat app is a complex task so the effort was done by the Status app team in the form of [Status Web](https://github.com/status-im/status-web/).
+
+Finally, (3) was the most important lesson.
```
Comment (oskarth) [inline research/2022-08-20-introducing-waku-product.md:65]: Maybe mention somewhere (not sure where) about what we learned in terms of separating out node operators, platforms and developers? From business workshop

File: research/2022-08-20-introducing-waku-product.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+
+# Waku for Your Project
+
+As discussed, Waku is now available on various platforms. The question remains: How can Waku benefit **your** project?
+
+Here are a couple of use cases we recently investigated:
+
+## Layer-2 Decentralization
+
+Most (all?) roll-ups use a centralized sequencer or equivalent. Running several sequencers is not as straightforward as running several execution nodes.
```
Comment (oskarth) [inline research/2022-08-20-introducing-waku-product.md:119]: Link to reference

**Review by oskarth — COMMENTED** [review]
Overall looks much better and almost there, just a few comments

File: research/2022-08-20-introducing-waku-product.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
+---
+layout: post
+name:  "Waku for All DApps"
```
[the remark this answers is not in the document]
Comment (liftlines) [reply research/2022-08-20-introducing-waku-product.md:3]: Agree with Oskar, lets call it 'Waku for all decentralised apps and infrastructure'.

File: research/2022-08-20-introducing-waku-product.md [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+title:  "Waku for All DApps"
+date:   2022-11-01 00:00:00 +0000
+author: franck
+published: true
+permalink: /waku-for-all-dapps
+categories: waku, dapp, platform, operator
+summary: Waku adoption push has shifted from Web to all platforms.
```
Comment (liftlines) [inline research/2022-08-20-introducing-waku-product.md:10]: Waku is an open communication protocol that decentralised apps and infrastructure can use for their communication needs. It is designed to enable dapps and decentralised infrastructure projects to access <add characteristics... e.g. scalable, highly performant, plug and play, ....> messaging features. It is accessible through <.... js Waku, nwaku,....> options.

File: research/2022-08-20-introducing-waku-product.md [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+  Goal:
+    - Trigger Ha-ha moment for Waku possibilities
+      - Explain shift from Web to all dApps
+    - Make it clear that Waku is being field tested, it is more than a research project
+    - Ideas on how to contribute/participate/play with Waku
+-->
+
+# Waku for the Web
+
+We are building Waku to be the communication layer for Web3:
```
Comment (liftlines) [inline research/2022-08-20-introducing-waku-product.md:26]: I'd turn this into past tense to show that it already exists and is here to be used. E.g. 'We have built Waku to be a communication layer for Web3'

File: research/2022-08-20-introducing-waku-product.md [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+a network and suite of protocols to enable secure, censorship-resistant, privacy-preserving communications that can run on any device.
```
Comment (liftlines) [inline research/2022-08-20-introducing-waku-product.md:27]: Could we instead say ' Waku network offers a collection of protocols to chose from for your messaging needs. It enables secure, censorship-resistant, spam-protected communication for its users.'

File: research/2022-08-20-introducing-waku-product.md [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+conversational security, protocol incentivization, zero-knowledge, anonymity to name a few.
+
+To ensure we reach our goals, we knew that early dogfooding of Waku was necessary, even if research was still _in progress_ [[1]](#footnote1).
+Thus, we started to push for the adoption of Waku by projects other than the Status app starting 2021.
+
+Waku is the communication component of the Web3 trifecta,
+which originally was Ethereum (contracts), Swarm (storage) and Whisper (communication).
+Hence, it made sense to first target dApps which already uses one of the Web3 pillars: Ethereum. 
+
+As most dApps are web apps,
```
[the remark this answers is not in the document]
Comment (liftlines) [reply research/2022-08-20-introducing-waku-product.md:39]: Agree, background section would be ideal grouping for this. Also, early on in this doc we should emphasise on readiness of this protocol for users, and subsequently talk about our continued research to improve it.

**Pull request merged**